types/mongodb/index.d.ts

```diff
@@ -1167,6 +1167,17 @@ export interface FindOneAndReplaceOption {
     session?: ClientSession;
 }
 
+/** http://mongodb.github.io/node-mongodb-native/3.1/api/Collection.html#findOneAndUpdate */
+export interface FindOneAndUpdateOption {
```

Comment:
You can extend `FindOneAndReplaceOption` adding only `arrayFilters`, to avoid the duplication